test/jest/roktManager.spec.ts

```diff
@@ -347,6 +416,122 @@ describe('RoktManager', () => {
         });
     });
 
+    describe('#generateMappedEventLookup', () => {
+        it('should generate a lookup table from a placement event mapping', () => {
+            const placementEventMapping: SettingMappingElement[] = [
+                {
+                    jsmap: '-1484452948',
+                    map: '-5208850776883573773',
+                    maptype: 'EventClass.Id',
+                    value: 'foo-mapped-flag',
+                },
+                {
+                    jsmap: '1838502119',
+                    map: '1324617889422969328',
+                    maptype: 'EventClass.Id',
+                    value: 'ad_viewed_test',
+                },
+            ];
+
+            const lookup = roktManager['generateMappedEventLookup'](placementEventMapping);
+            expect(lookup).toEqual({
+                '-1484452948': 'foo-mapped-flag',
+                '1838502119': 'ad_viewed_test'
+            });
+        });
+        
+        it('should return an empty object if the placement event mapping is empty', () => {
+            const lookup = roktManager['generateMappedEventLookup']([]);
+            expect(lookup).toEqual({});
+        });
```

Comment:
maybe add a test that puts an invalid mapping in here to see if the error handling is robust?